trieTools/trieChecker/logParser/formatter.go

```diff
@@ -0,0 +1,53 @@
+package logParser
+
+import (
+	"strings"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+// TODO remove this package
+
+type loggerFormatter struct {
+	mutErrors    sync.RWMutex
+	errorStrings []string
+}
+
+// NewLoggerFormatter creates a new logger formatter instance
+func NewLoggerFormatter() *loggerFormatter {
+	return &loggerFormatter{
+		errorStrings: make([]string, 0),
+	}
+}
+
+// Output will always return nil byte slice but records any type of errors encountered
+func (formatter *loggerFormatter) Output(line logger.LogLineHandler) []byte {
+	if line.GetLogLevel() != int32(logger.LogError) {
+		return nil
+	}
+
+	data := append([]string{line.GetMessage()}, line.GetArgs()...)
+
+	formatter.mutErrors.Lock()
+	formatter.errorStrings = append(formatter.errorStrings, strings.Join(data, " "))
```

Comment:
Nice error interceptor :)

Reply:
it's a terrible hack, but the only one possible, for now :(